lib/net.js

```diff
@@ -933,14 +943,22 @@ Socket.prototype.connect = function(options, cb) {
 
 
 Socket.prototype.ref = function() {
-  if (this._handle)
-    this._handle.ref();
+  if (!this._handle) {
+    this.once('connect', this.ref.bind(this));
```

Comment:
The `.bind(this)` is superfluous, just `this.once('connect', this.ref)` will do.